scaleway/resource_mnq_credential.go

```diff
@@ -0,0 +1,268 @@
+package scaleway
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	mnq "github.com/scaleway/scaleway-sdk-go/api/mnq/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	defaultMNQCredentialTimeout = 5 * time.Minute
+)
+
+func resourceScalewayMNQCredential() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayMNQCredentialCreate,
+		ReadContext:   resourceScalewayMNQCredentialRead,
+		UpdateContext: resourceScalewayMNQCredentialUpdate,
+		DeleteContext: resourceScalewayMNQCredentialDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create:  schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Read:    schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Update:  schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Delete:  schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Default: schema.DefaultTimeout(defaultMNQCredentialTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				Description: "The name of the Credential",
+			},
+			"namespace_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The ID of the Namespace associated to",
+			},
+			// computed
+			"region": regionSchema(),
+			"protocol": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The Namespace protocol",
+			},
+			"nats_credentials": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"sqs_sns_credentials"},
+				Description:   "credential for NATS protocol",
+				MaxItems:      1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"content": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Sensitive:   true,
+							Description: "Raw content of the NATS credentials file",
+						},
+					},
+				},
+			},
+			"sqs_sns_credentials": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				Description:   "The credential used to connect to the SQS/SNS service",
+				MaxItems:      1,
+				ConflictsWith: []string{"nats_credentials"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"permissions": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							MaxItems:    1,
+							Description: "The permission associated to this credential.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"can_publish": {
+										Type:        schema.TypeBool,
+										Default:     true,
```

Comment:
`can_publish`, `can_receive` and `can_manage` should be `false` by default (nullable)